packages/pkg/src/index.ts

```diff
@@ -13,16 +13,28 @@ import babelParser from 'prettier/plugins/babel.js'
 import { files } from './rules/files.js'
 import { object } from './rules/object.js'
 import { dependencyNames, sort } from './rules/sort.js'
-import type { ObjectExpression, ObjectProperty } from './types.js'
+import type {
+  ObjectExpression,
+  ObjectProperty,
+  FormatOptions,
+} from './types.js'
 
 const PKG_REG = /[/\\]?package\.json$/
 
+export const UNTS_SORT_ORDER: readonly string[] = [
+  'engines',
+  'devEngines',
+  'scripts',
+  ...dependencyNames,
+]
+
 const {
   json: { parse },
 } = babelParser.parsers
 
-const format = (properties: ObjectProperty[]) => {
-  let props = ['engines', 'devEngines', 'scripts', ...dependencyNames].reduce(
```

Comment:
It seems incorrect, they're only for sortable **properties** like `engines`, `devEngines `, etc. not the top level order.

We should change `sort(properties)` instead.

Reply:
It's just not working as you expected. The sorting happens at `sort(properties)`, not here.

Reply:
😲 👌